### DISCUSSION TITLE
How can i setup Pairdrop with Caddy in home-lab environment

### DISCUSSION THREAD

**Discussion by cesare190** [schlagmichdoch/PairDrop#242] [Q&A]
Good morning, i'm trying to setup Caddy as reverse-proxy for Pairdrop. At the moment Pairdrop is a Docker Container in a custom bridge that expose port binded to host i.e. 127.0.0.1:5200 -> 3000. I also have a Caddy docker container that work on host network that have a configuration to expose Pairdrop to Home network only (General idea should be <fqdn> (192.168.x.x:443) -> 127.0.0.1:5200 -> pairdrop:3000). Caddy manage https with FQDN Let's Encrypt certificate that i also use for my homelab.

I tried to follow guides and tutorial to setup a reverse proxy: with Nginx Proxy Manager I didn't have Problem but insted i'm not able to manage Caddy. I suppose that the problem is with ws but I can't find how to correctly manage them in Caddy.

Thanks so much.

**Comment by cesare190** [comment]
Is it correct?

http://REDACTED {
        redir https://{host}{uri}
}

https://REDACTED {
        import REMOVED-CERT-FOR-SSL
        reverse_proxy * http://127.0.0.1:choosenport {
                header_up Host {upstream_hostport}
        }
}

Or there is more that i have to do?
With * seems to go, but with / in reverse_proxy it seems broken

**Comment by ottsimon** [comment]
Hi, I started using PairDrop together with Caddy V2 as a reverse proxy.

My Caddyfile looks like this:
```
pairdrop.example.com {
	reverse_proxy 127.0.0.1:3000
}
```

Some notes:
- Make sure to change the domain and port accordingly to your setup.
- If you use IPv6, make sure `IPV6_LOCALIZE` is set (see [documentation](https://github.com/schlagmichdoch/PairDrop/blob/master/docs/host-your-own.md#ipv6-localization)).
- You don't need to specify a http to https redirect, Caddy does this automatically for you.
- You also don't need to specify any headers, Caddy will set the required ones automatically too.
- If you want to use http only, specify this in front of the domain: `http://pairdrop.example.com`.
- If you want to use custom certificates, you can specify them inside the domain block with `tls /path/to/cert /path/to/key`.
- Otherwise, Caddy will get a Let's Encrypt certificate for you.

Hope that helps.
